src/main/scala/io/iohk/ethereum/network/EtcMessageHandler.scala

```diff
@@ -0,0 +1,153 @@
+      initialPeerInfo.withTotalDifficulty(newBlock.totalDifficulty)
+    case _ => initialPeerInfo
+  }
+
+  /**
+    * Processes the message and updates if the fork block was accepted from the peer
+    *
+    * @param message to be processed
+    * @param initialPeerInfo from before the message was processed
+    * @return new peer info with the fork block accepted value updated
+    */
+  private def updateForkAccepted(message: Message)(initialPeerInfo: EtcPeerInfo): EtcPeerInfo = message match {
+    case BlockHeaders(blockHeaders) =>
+      val newPeerInfoOpt: Option[EtcPeerInfo] =
+        for {
+          forkResolver <- forkResolverOpt
+          forkBlockHeader <- blockHeaders.find(_.number == forkResolver.forkBlockNumber)
+        } yield {
+          val newFork = forkResolver.recognizeFork(forkBlockHeader)
+          log.info("Received fork block header with fork: {}", newFork)
+
+          if (!forkResolver.isAccepted(newFork)) {
+            log.warn("Peer is not running the accepted fork, disconnecting")
+            peer.ref ! DisconnectPeer(Disconnect.Reasons.UselessPeer)
+            initialPeerInfo
+          } else
+            initialPeerInfo.withForkAccepted(true)
+        }
+      newPeerInfoOpt.getOrElse(initialPeerInfo)
+
+    case _ => initialPeerInfo
+  }
+
+  /**
+    * Processes the message and updates the max block number from the peer
+    *
+    * @param message to be processed
+    * @param initialPeerInfo from before the message was processed
+    * @return new peer info with the max block number updated
+    */
+  private def updateMaxBlock(message: Message)(initialPeerInfo: EtcPeerInfo): EtcPeerInfo = {
+    def update(ns: Seq[BigInt]): EtcPeerInfo = {
+      val maxBlockNumber = ns.fold(0: BigInt) { case (a, b) => if (a > b) a else b }
+      if (maxBlockNumber > appStateStorage.getEstimatedHighestBlock())
+        appStateStorage.putEstimatedHighestBlock(maxBlockNumber)
+
+      if (maxBlockNumber > initialPeerInfo.maxBlockNumber)
+        initialPeerInfo.withMaxBlockNumber(maxBlockNumber)
+      else
+        initialPeerInfo
+    }
+
+    message match {
+      case m: BlockHeaders =>
+        update(m.headers.map(_.number))
+      case m: NewBlock =>
+        update(Seq(m.block.header.number))
+      case m: NewBlockHashes =>
+        update(m.hashes.map(_.number))
+      case _ => initialPeerInfo
+    }
+  }
+
+  /**
+    * Processes the message and updates if the fork block was accepted from the peer
+    *
+    * @param message to be processed
+    * @param initialPeerInfo from before the message was processed
+    * @return new peer info with the fork block accepted value updated
+    */
+  private def handleSendingNewBlock(message: Message, initialPeerInfo: EtcPeerInfo): MessageAction = message match {
+    case newBlock: NewBlock =>
+      if (newBlock.block.header.number > initialPeerInfo.maxBlockNumber ||
+        newBlock.totalDifficulty > initialPeerInfo.totalDifficulty)
```

Comment:
You are right, it makes sense to send this blocks so that they can be included as ommers :+1:
Regarding that, I'm wondering, should we also send new blocks that have a low block number so that they are considered for being ommers? For example geth checks the block number to see if it is a possible ommer (https://github.com/ethereumproject/go-ethereum/blob/master/eth/fetcher/fetcher.go#L312), however that would require changing how we store the known blocks by a peer (instead of only using the max block number) to something similar to how we handle known txs. What do you think about this?